addons/mail/models/mail_message.py

```diff
@@ -1108,7 +1121,7 @@ def _message_format_personalized_prepare(self, messages_formatted, partner_ids=N
             vals['follower_id_by_partner_id'] = followers_by_record_ref.get((vals['model'], vals['res_id']), dict())
         return messages_formatted
 
-    def _message_format_personalize(self, partner_id, messages_formatted=None, format_reply=True, msg_vals=None):
+    def _message_format_personalize(self, partner_id, user_id=None, messages_formatted=None, format_reply=True, msg_vals=None):
```

Comment:
ok, I checked the commit and we did something similar, only if the uid is got from `(r['id'], r['uid']) for r in recipients_data`, it should be easy to backport without adding extra queries. Is your PR coming soon? I guess it will be easier for me to redo this PR when your PR is merged and backported.

Reply:
I'm not going to backport my PR, you have to take the interesting part and apply them here.

I copied some of your work here to make it, just I went further with cleaning the methods.

I think runbot should be green, there is no query increase.

Note: commit was updated to add missing sudo and convert also mock server: https://github.com/odoo-dev/odoo/commit/5401b3f35c6feaeabe22d65c6c0820b62098a23a